dispersy.py

```
@@ -2909,14 +2766,14 @@ def _forward(self, messages):
         elif isinstance(meta.destination, MemberDestination):
             # MemberDestination.candidates may be empty
             result = all(self._send([candidate
-                                            for candidate
-                                            in self._candidates.itervalues()
-                                            if any(candidate.is_associated(message.community, member)
-                                                   for member
-                                                   in message.destination.members)],
-                                           [message])
-                       for message
-                       in messages)
+                                     for candidate
+                                     in message.community.candidates.itervalues()
```

Ok great, it's a bit scary now as you need to keep the candidates you're going to send a signature request message to active without screwing with the walker. That seems like an impossible task.

Maybe we can add a return false if no candidates were found for any of these members. As this could be a silent fail, which is occurring in the bartercast community now.